cps/kni.c

```diff
@@ -295,6 +299,345 @@ kni_config(struct rte_kni *kni, struct gatekeeper_if *iface)
 	return ret;
 }
 
+void
+route_event_sock_close(struct cps_config *cps_conf)
+{
+	if (cps_conf->nl != NULL) {
+		mnl_socket_close(cps_conf->nl);
+		cps_conf->nl = NULL;
+	}
+}
+
+int
+route_event_sock_open(struct cps_config *cps_conf)
+{
+	struct mnl_socket *nl;
+	int ret;
+
+	nl = mnl_socket_open(NETLINK_ROUTE);
+	if (nl == NULL) {
+		RTE_LOG(ERR, GATEKEEPER, "cps: %s: mnl_socket_open: %s\n",
+			__func__, strerror(errno));
+		return -1;
+	}
+	cps_conf->nl = nl;
+
+	ret = mnl_socket_bind(nl, RTMGRP_IPV4_ROUTE|RTMGRP_IPV6_ROUTE,
```

Comment:
Please add the following TODO comment before `mnl_socket_bind()`.
```
/* TODO This bind will get all changes that arrive at the kernel,
 * but it will duplicate the FIB: one in the kernel and another in Gatekeeper.
 */
```